### ISSUE TITLE
API  addic7ed.com changed, tool no langer works

### ISSUE THREAD

**Issue by mister131** [phdelodder/SubTools#90]

**Comment by Wally73** [comment]
you mean api of thetvbd changed (addicted hasn't have an api (yet, they are working on it though)

**Comment by Denniswolt** [comment]
Is it possible to fix this?

**Comment by Wally73** [comment]
by more investigation its due the fact that addicted has implemented recaptcha v2 for logins

**Comment by Denniswolt** [comment]
So that means this tool no longer works, or can this be fixed?

**Comment by TheFonZzz** [comment]
Same kind of error with Opensubtitles.
Hopefully someone can help the maker to solve these problems.

**Comment by Denniswolt** [comment]
Can somebody make it work again please? Or point to alternatives?

**Comment by EotT123** [comment]
Fixed in https://github.com/phdelodder/SubTools/pull/93, although addic7ed is very slow.

**Comment by TheFonZzz** [comment]
> Fixed in #93, although addic7ed is very slow.

Thank you very much for picking up this problem after a long, long time.
Unfortunatly it doesn't work here, I keep on getting an INI error... Maybe I've got to reinstall JAVA.

**Comment by TheFonZzz** [comment]
I did reinstalled JAVA with the latest version, but still I keep on getting an INI error. Anything I can do to fix this problem ??

**Comment by EotT123** [comment]
You need the latest java version (java 18). I use the latest OpenJdk, which you can download here: https://jdk.java.net/18/. But I don't think this is the problem.
Can you describe the error you are getting a bit more? Is there any more info?

**Comment by TheFonZzz** [comment]
I use Java build 1.8.0_341-b10.
The error I get in the pop-up screen is:
"Error, a INI Errror has occurred, please check your installation and try again"
"A Java exception has occurred".

The last version of Multisubdownloader (13.0 with dependecies) works still fine (offcourse with the known API errors like reported above).

**Comment by EotT123** [comment]
You are using java 8, not 18.

**Comment by TheFonZzz** [comment]
Java 8 is the latest (consumer) version. The JDK version is indeed at version 18, but I don't know whether it is a wise thing to install a Developer verion on a regular PC.

**Comment by EotT123** [comment]
You don't need the jdk indeed, because it includes many things you don't need. But it isn't 'bad' either.
In case you don't want to download the full jdk package, there are much smaller jre packages available here: https://adoptium.net/temurin/releases/?version=18

FYI: In a few days, java 19 will be released. I'm planning to update to this version, to be able to use new features. Those new feature aren't important for a normal user, but you can still benefit from the optimizations regaring performance and memory usage.

**Comment by TheFonZzz** [comment]
I installed the JDK version, but I still keep on getting errors like:
[ERROR] createRelease
[ERROR] Unknown serie name in tvdb: The Rookie
Only one serie and one movie are showing up.

**Comment by Denniswolt** [comment]
I have the exact same error.
The error I get in the pop-up screen is:
"Error, a INI Errror has occurred, please check your installation and try again"
"A Java exception has occurred".

How can I run 1.4.0? I'm using Java 1.8.0_341-b10.

**Comment by EotT123** [comment]
> [ERROR] createRelease [ERROR] Unknown serie name in tvdb: The Rookie Only one serie and one movie are showing up.
This will probably be fixed in the new version. I've created a new PR: https://github.com/phdelodder/SubTools/pull/108

**Comment by Denniswolt** [comment]
When downloading from opensubtitles I get an error:
[INFO] Downloading subtitle:
[Bull.2016.S06E01.iNTERNAL.1080p.WEB.H264-WHOSNEXT.srt] for release:
[bull.2016.s06e01.internal.720p.web.h264-whosnext.mkv]
[ERROR] org.opensubtitles.invoker.ApiException: Message:
HTTP response code: 406
HTTP response body: {"requests":8,"remaining":-3,"message":"You have
downloaded your allowed 5 subtitles for 24h.Your quota will be renewed in
23 hours and 00 minutes (2022-09-28 19:58:25 UTC) ","reset_time":"23 hours
and 00 minutes","reset_time_utc":"2022-09-28T19:58:25.000Z"}
HTTP response headers: {access-control-allow-origin=[*], age=[0],
alt-svc=[h3=":443"; ma=86400, h3-29=":443"; ma=86400],
cache-control=[max-age=0, private, must-revalidate, s-maxage=600],
cf-cache-status=[DYNAMIC], cf-ray=[7517279659670c71-AMS],
content-length=[258], content-type=[application/json; charset=utf-8],
date=[Tue, 27 Sep 2022 20:57:58 GMT], expires=[0], last-modified=[Tue, 27
Sep 2022 20:57:58 GMT],
nel=[{"success_fraction":0,"report_to":"cf-nel","max_age":604800}],
pragma=[no-cache], ratelimit-limit=[5], ratelimit-remaining=[4],
ratelimit-reset=[1], referrer-policy=[strict-origin-when-cross-origin],
report-to=[{"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=ig%2FcanElMWck2qSsozwJaaSJljLDHHOu09ynU61pg9QYnvdW7jhBr0LCAebiETbKiGsveoEOCLUmknCvSUDmXz4wCBU26yyoFvcT4GqRJgprZdBvJQbGes74D3%2Fx7xf38RGLwVlhy7c%3D"}],"group":"cf-nel","max_age":604800}],
server=[cloudflare], strict-transport-security=[max-age=15552000;
includeSubDomains; preload], x-cache-backend=[apigw1_8000 rb1],
x-content-type-options=[nosniff], x-download-options=[noopen],
x-frame-options=[SAMEORIGIN], x-kong-proxy-latency=[1],
x-kong-upstream-latency=[514], x-permitted-cross-domain-policies=[none],
x-ratelimit-limit-second=[5], x-ratelimit-remaining-second=[4],
x-request-id=[b1889bc7-d48c-4080-b2f9-a6eabd6340f1], x-runtime=[0.114848],
x-var-cache=[MISS], x-via=[fw1], x-xss-protection=[1; mode=block]}

Is this something new? Never happend on with older versions.

Op zo 25 sep. 2022 om 15:37 schreef EotT123 ***@***.***>:

**Comment by EotT123** [comment]
To solve this, you have to create an account on opensubtitle and add the credentials in the settings menu. Otherwise, you'll be limited to download 5 subtitles per 24 hours. (see https://github.com/phdelodder/SubTools/pull/91). The old api is deprecated and will be removed (by opensubtitles, so there is nothing I can do about that).

**Comment by Denniswolt** [comment]
> To solve this, you have to create an account on opensubtitle and add the credentials in the settings menu. Otherwise, you'll be limited to download 5 subtitles per 24 hours. (see #91). The old api is deprecated and will be removed (by opensubtitles, so there is nothing I can do about that).

I did that, but still the same errror

**Comment by EotT123** [comment]
Which version are you using? The latest released version (1.4.0)? Or did you build a version yourself based on the latest code? It's possible that there was a problem which is already fixed. 
I've attached a new version which is based on the latest code in case you can't build it yourself: [multisubdownloader-1.4.1-SNAPSHOT-jar-with-dependencies.zip](https://github.com/phdelodder/SubTools/files/9692672/multisubdownloader-1.4.1-SNAPSHOT-jar-with-dependencies.zip)

**Comment by Denniswolt** [comment]
I was using 1.4.0. I don't know how to build a version myself. Thank you for 1.4.1. Seems to be working perfect. Except it disregards the excluded folders to scan.

**Comment by TheFonZzz** [comment]
I tried the 1.4.1 version and it looks very nice (great option with the automatic search for series). But I get dozens of this ERROR line: 

_[ERROR] API OpenSubtitles searchSubtitles for serie [Day of the Dead S01E04] (Cannot invoke "org.lodder.subtools.multisubdownloader.subtitleproviders.opensubtitles.api.v2.OpenSubtitlesApi.searchSubtitles()" because the return value of "org.lodder.subtools.multisubdownloader.subtitleproviders.adapters.JOpenSubAdapter.getApi()" is null)_

**Comment by EotT123** [comment]
I'm working on a new version, but it takes some time.
@Denniswolt  Excluded folders is indeed bugged, I'll try to fix it in the new version.
@TheFonZzz I also encountered that error, but it disappeared when I tried it a second time. I'll also have a look at this.

**Comment by TheFonZzz** [comment]
> I'm working on a new version, but it takes some time. @Denniswolt Excluded folders is indeed bugged, I'll try to fix it in the new version. @TheFonZzz I also encountered that error, but it disappeared when I tried it a second time. I'll also have a look at this.

Great news, thank you for your work.
I know it can take a lot of time, for making a new version. We will wait with patient..

**Comment by EotT123** [comment]
A new version is ready, I've created a new PR: https://github.com/phdelodder/SubTools/pull/115

@Denniswolt Exclude folders should be fixed.
@TheFonZzz I'm not 100% sure that the error is gone. Please try the new version and let me know if you still encounter that error.

If you don't want to wait until a new version is released, you could already try this one: 
[multisubdownloader-1.5.1-SNAPSHOT-jar-with-dependencies.zip](https://github.com/phdelodder/SubTools/files/9844393/multisubdownloader-1.5.1-SNAPSHOT-jar-with-dependencies.zip)

**Comment by TheFonZzz** [comment]
Thank you @EotT123 for this new version.
It starts good and it runs fine till it stuck at one serie. I couldn't find an option to add this one manual which might solve the problem..
Here's the part of the logging from where it gone wrong:
_[ERROR] createRelease
[TRACE] Parsing match found using file name: Doc.Martin.S010E05.1080p.STV.WEB-DL.AAC2.0.H.264[eztv.re].avi
[TRACE] extractReleasegroup: name: Doc.Martin.S010E05.1080p.STV.WEB-DL.AAC2.0.H.264[eztv.re].avi , hasExtension: true
[TRACE] extractReleasegroup: release group: 
[TRACE] getQualityKeyword: name: Doc.Martin.S010E05.1080p.STV.WEB-DL.AAC2.0.H.264[eztv.re].avi
[TRACE] getQualityKeyWords: keyswords: 1080p WEB-DL_

**Comment by EotT123** [comment]
@TheFonZzz  This is still original code of phdelodder, I haven't touched that part. I could try to improve it, but for now, you could rename your file to _Doc.Martin.S10E05.1080p.STV.WEB-DL.AAC2.0.H.264[eztv.re].avi_ (S**0**10E05 -> S10E05).

**Comment by Denniswolt** [comment]
[ERROR] API Addic7ed searchSubtitles for serie [New Amsterdam (2018) S05E04] (Exception while getting value (org.lodder.subtools.sublibrary.ManagerException: Error occured with httpclient response: 302 Moved Temporarily))
[ERROR] API Addic7ed searchSubtitles for serie [New Amsterdam (2018) S05E05] (Exception while getting value (org.lodder.subtools.sublibrary.ManagerException: Error occured with httpclient response: 302 Moved Temporarily))
[ERROR] API Addic7ed searchSubtitles for serie [New Amsterdam (2018) S05E06] (Exception while getting value (org.lodder.subtools.sublibrary.ManagerException: Error occured with httpclient response: 302 Moved Temporarily))
[ERROR] API Addic7ed searchSubtitles for serie [S.W.A.T. (2017) S06E01] (Exception while getting value (org.lodder.subtools.sublibrary.ManagerException: Error occured with httpclient response: 302 Moved Temporarily))
[ERROR] API Addic7ed searchSubtitles for serie [S.W.A.T. (2017) S06E02] (Exception while getting value (org.lodder.subtools.sublibrary.ManagerException: Error occured with httpclient response: 302 Moved Temporarily))
[ERROR] API Addic7ed searchSubtitles for serie [S.W.A.T. (2017) S06E03] (Exception while getting value (org.lodder.subtools.sublibrary.ManagerException: Error occured with httpclient response: 302 Moved Temporarily))
[ERROR] API Addic7ed searchSubtitles for serie [SEAL Team S06E05] (Exception while getting value (org.lodder.subtools.sublibrary.ManagerException: Error occured with httpclient response: 302 Moved Temporarily))

Addic7ed does not seem to be working.

**Comment by EotT123** [comment]
You can try the Addic7ed proxy (can be selected in _Serie providers_ tab in settings)

**Comment by TheFonZzz** [comment]
This is an error I see since working with the latest version (1.5.1)

_[ERROR] API OpenSubtitles searchSubtitles for serie [Yellowstone (2018) S05E08] (org.lodder.subtools.multisubdownloader.subtitleproviders.opensubtitles.exception.OpenSubtitlesException: org.opensubtitles.invoker.ApiException: Message: 
HTTP response code: 429
HTTP response body: {
  "message":"API rate limit exceeded"_

**Comment by EotT123** [comment]
@TheFonZzz  OpenSubtitles is blocking your request, because you have done too many requests. 
This case is handled by the code: it waits 5 seconds before trying again. But apparently, OpenSubtitles is still blocking the requests. I could not reproduce this behavior. Does this happen at the first subtitle you are searching with OpenSubtitles, or only after a while? Is it solved if you try again later?

Btw, 1.5.1 is not yet released, new 1.5.1 versions are build automatically when new features or fixes are added. You can find the latest nightly build here: https://github.com/phdelodder/SubTools/actions . The top row contains the latest build. The new version will include automatic/manual checking for new stable and nightly versions.

**Comment by TheFonZzz** [comment]
@EotT123 It happens after a little while. Maybe because I've got so many titles for which I'm looking for subs (about 95).
It doesn't look like it waits 5 seconds before trying again. In that case it would take a much, much longer time before  the program would be ready to search. But after about a minute it is already done.

I'll try that 1.5.0 version later today..

Thanks for all your work. I really appreciate.

**Comment by EotT123** [comment]
@TheFonZzz The 5 seond delay should only start when a request is blocked. So maybe 20 request can be done, followed by a wait of 5 seconds, before a new batch of 20 request can be done. So you'll only have to wait 30 seconds for the 95 requests. (These numbers are just to give you an example, I don't know how many requests can be done before a new request is blocked).  
But it also possible that after a while and a bunch of requests, you have to wait longer than 5 sedonds. I implemented a retry mechanism, which will try again after 5 seconds, 3 times in a row. If the request is still blocked after all those retries, it will fail and is skipped (and you'll see an error message).
